src/e2e/global-teardown.ts

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import https from "https";
+import { readFileSync, writeFileSync } from "fs";
+import { CustomTestResult } from "../../E2EReporter";
+
+async function globalTearDown() {
+  // check env for cron, teardown is only needed for teardown run
+  if (process.env.IS_CRON) {
```

Comment:
I might invert your logic here (and L45-47 below) so you check if `IS_CRON` _isn't_ true and just abort early:
```js
if (process.env.IS_CRON !== "true") {
  console.log("[e2e_log] - Not a cron json, json report will not be created");
  return;
}
```
That way we lose a depth of nesting and the `else` bit and keeps is slightly flatter.